extensions/jdbc/jdbc-oracle/deployment/src/main/java/io/quarkus/jdbc/oracle/deployment/OracleMetadataOverrides.java

```diff
@@ -142,24 +143,24 @@ void excludeOracleDirectives(BuildProducer<ExcludeConfigBuildItem> nativeImageEx
                 .produce(new ExcludeConfigBuildItem(DRIVER_JAR_MATCH_REGEX, NATIVE_IMAGE_REFLECT_CONFIG_MATCH_REGEX));
     }
 
-    @BuildStep
+    @BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)
     NativeImageAllowIncompleteClasspathBuildItem naughtyDriver() {
         return new NativeImageAllowIncompleteClasspathBuildItem("quarkus-jdbc-oracle");
     }
 
-    @BuildStep
+    @BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)
     RemovedResourceBuildItem overrideSubstitutions() {
         return new RemovedResourceBuildItem(ArtifactKey.fromString("com.oracle.database.jdbc:ojdbc11"),
                 Collections.singleton("oracle/nativeimage/Target_java_io_ObjectStreamClass.class"));
```

Comment:
The `oracle/nativeimage/Target_java_io_ObjectStreamClass.class` is not part of the most recent Oracle JDBC Driver 23.2.0.0. Should it be removed, or is there any support for backward-compatible JDBC drivers?